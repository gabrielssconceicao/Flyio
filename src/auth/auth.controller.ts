import { Body, Controller, HttpCode, HttpStatus, Post } from '@nestjs/common';
import { ApiBody, ApiOperation, ApiResponse, PickType } from '@nestjs/swagger';
import { AuthService } from './auth.service';
import { LoginDto, LoginResponseDto, RefreshTokenDto } from './dto';

@Controller('auth')
export class AuthController {
  constructor(private readonly authService: AuthService) {}

  @ApiOperation({ summary: 'User Login ' })
  @ApiBody({
    type: LoginDto,
    description: 'User Login',
  })
  @ApiResponse({
    status: HttpStatus.CREATED,
    description: 'Token',
    type: LoginResponseDto,
  })
  @ApiResponse({
    status: HttpStatus.UNAUTHORIZED,
    description: 'User or password invalid',
    schema: {
      example: {
        statusCode: HttpStatus.UNAUTHORIZED,
        message: 'User or password invalid',
        error: 'Unauthorized',
      },
    },
  })
  @ApiResponse({
    status: HttpStatus.FORBIDDEN,
    description: 'User not active',
    schema: {
      example: {
        statusCode: HttpStatus.FORBIDDEN,
        message: 'User not active',
        error: 'Forbidden',
      },
    },
  })
  @HttpCode(HttpStatus.CREATED)
  @Post('login')
  login(@Body() loginDto: LoginDto) {
    return this.authService.login(loginDto);
  }

  @ApiOperation({ summary: 'Refresh access token' })
  @ApiBody({
    description: 'Refresh token',
    type: RefreshTokenDto,
  })
  @ApiResponse({
    status: HttpStatus.CREATED,
    description: 'New access token',
    type: PickType(LoginResponseDto, ['accessToken']),
  })
  @ApiResponse({
    status: HttpStatus.FORBIDDEN,
    description: 'Token expired',
    schema: {
      example: {
        statusCode: HttpStatus.FORBIDDEN,
        message: 'Token expired',
        error: 'Forbidden',
      },
    },
  })
  @HttpCode(HttpStatus.CREATED)
  @Post('refresh-token')
  refreshToken(@Body() refreshTokendto: RefreshTokenDto) {
    return this.authService.refreshToken(refreshTokendto);
  }
}
