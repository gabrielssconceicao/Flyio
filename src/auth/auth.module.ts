import { Global, Module } from '@nestjs/common';
import { BcryptService } from './bcrypt.service';
import { HashingServiceProtocol } from './hashing/hashing.service';

@Global()
@Module({
  imports: [],
  providers: [
    {
      provide: HashingServiceProtocol,
      useClass: BcryptService,
    },
  ],
  exports: [HashingServiceProtocol],
})
export class AuthModule {}
