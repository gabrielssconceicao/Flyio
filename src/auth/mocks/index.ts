import { hashingServiceMock } from './hashing.service.mock';
import { jwtConfigurationMock } from './jwt.configuration.mock';
import { jwtServiceMock } from './jwt.service.mock';
import { generateLoginDtoMock } from './login.dto.mock';
import { generateTokenPayloadDtoMock } from './token-payload.dto.mock';

export {
  hashingServiceMock,
  jwtConfigurationMock,
  jwtServiceMock,
  generateLoginDtoMock,
  generateTokenPayloadDtoMock,
};
