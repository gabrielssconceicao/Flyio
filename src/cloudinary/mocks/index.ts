import { cloudinaryServiceMock } from './cloudinary.service.mock';
import { generateFileMock, generatedProfilePictureMock } from './file.mock';

export { cloudinaryServiceMock, generateFileMock, generatedProfilePictureMock };
