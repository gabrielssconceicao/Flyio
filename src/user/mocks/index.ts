import { generateUserMock } from './user.mock';
import { generateFindAllUsersResponseDtoMock } from './generate-find-all-users-response.dto.mock';
import { generateCreateUserDtoMock } from './generate-create-user.dto.mock';

export {
  generateUserMock,
  generateFindAllUsersResponseDtoMock,
  generateCreateUserDtoMock,
};
