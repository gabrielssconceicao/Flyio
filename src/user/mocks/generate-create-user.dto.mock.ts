import { CreateUserDto } from '../dto';
import { userMock } from './user.mock';
export const generateCreateUserDtoMock = (
  bio: boolean = false,
): CreateUserDto => {
  const dto: CreateUserDto = {
    name: userMock.name,
    username: userMock.username,
    email: userMock.email,
    password: userMock.password,
  };

  if (bio) dto.bio = userMock.bio;
  return dto;
};
