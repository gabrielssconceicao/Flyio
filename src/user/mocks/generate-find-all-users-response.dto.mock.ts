import { FindAllUsersResponseDto } from '../dto';
import { userMock } from './user.mock';

export const generateFindAllUsersResponseDtoMock =
  (): FindAllUsersResponseDto => {
    return {
      count: 1,
      items: [
        {
          id: userMock.id,
          name: userMock.name,
          username: userMock.username,
          profileImg: userMock.profileImg,
        },
      ],
    };
  };
