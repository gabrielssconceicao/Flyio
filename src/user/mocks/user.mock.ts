import { User } from '../entities/user.entity';

export const userMock = {
  id: '42-d-f-df4',
  name: 'John Doe',
  username: 'jDoe453',
  email: 'jdoe@me.com',
  password: '123456',
  profileImg: 'https://example.com/profile.jpg',
  bio: 'This is my bio',
  active: true,
};
export const generateUserMock = (): User => {
  return {
    id: userMock.id,
    name: userMock.name,
    username: userMock.username,
    profileImg: userMock.profileImg,
    bio: userMock.bio,
    active: userMock.active,
    followers: 0,
    following: 0,
  };
};
