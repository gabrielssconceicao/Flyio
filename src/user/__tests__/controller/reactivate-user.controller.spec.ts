import { Test, TestingModule } from '@nestjs/testing';
import { JwtService } from '@nestjs/jwt';
import jwtConfig from 'src/auth/config/jwt.config';
import { jwtServiceMock, jwtConfigurationMock } from 'src/auth/mocks';

import { ReactivateUserService } from '../../reactivate-user.service';
import { ReactivateUserController } from 'src/user/reactivate-user.controller';

describe('UserController', () => {
  let controller: ReactivateUserController;
  let service: ReactivateUserService;

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      controllers: [ReactivateUserController],
      providers: [
        {
          provide: ReactivateUserService,
          useValue: {
            reactivate: jest.fn(),
          },
        },

        {
          provide: JwtService,
          useValue: jwtServiceMock,
        },
        {
          provide: jwtConfig.KEY,
          useValue: jwtConfigurationMock,
        },
      ],
    }).compile();

    controller = module.get<ReactivateUserController>(ReactivateUserController);
    service = module.get<ReactivateUserService>(ReactivateUserService);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should be defined', () => {
    expect(controller).toBeDefined();
    expect(service).toBeDefined();
  });

  it('should reactivate a user successfully', async () => {
    const reactivateUserDto = {
      token: 'token',
    };
    const message = 'User reactivated successfully';
    jest.spyOn(service, 'reactivate').mockResolvedValue({ message });
    const result = await controller.reactivate(reactivateUserDto);
    expect(service.reactivate).toHaveBeenCalledWith(reactivateUserDto);
    expect(result).toEqual({ message });
    expect(result).toMatchSnapshot();
  });
});
