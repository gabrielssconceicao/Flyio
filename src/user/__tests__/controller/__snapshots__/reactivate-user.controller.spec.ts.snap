// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserController should reactivate a user successfully 1`] = `
{
  "message": "User reactivated successfully",
}
`;
